Try three times to get a tweet before raising an error.

progval opened the pull request.

progval committed:
Try three times to get a tweet before raising an error.

RaitoBezarius commented on backend/realtime_worker.py:
```diff
@@ -23,7 +26,8 @@ def fetch_tweets(db, stream):
         if isTransportOffer(tweet):
             print ('Adding tweet from @{}'.format(tweet['user']['screen_name']))
-            r.db('voyageavecmoi').table('offers').insert(get_tweet_data(tweet)).run(db)
+            tweet_data = retry_call(get_tweet_data, (tweet,), tries=MAX_TRIES)
```
Why do you retry to extract tweet data if you already have the data?
The `get_tweet_data` function is somehow side-effect free, right?

progval replied:
Hmm, good point, the issue was with `stream.statuses.filter(track='#VoyageAvecMoi')` actually (but I could not reproduce the error from Twitter so I couldn't test it)

RaitoBezarius replied:
Alright, I think that we should wrap the iteration in a `try/catch` and make somehow restart the loop.

progval commented:
Fixed the issue you pointed out by retrying the actual failing line (the for loop)

progval commented:
Note that now it will try indefinitely, with an exponential backoff

progval committed:
Fix previous commit by retrying what should actually be retried.

progval committed:
Add a confirmation when a tweet is added.

progval committed:
Add a logger.

progval commented:
Actually, I am not sure this is the right way, because this does not report success to the library, so even after a success, the backoff will be increased.
Any better idea?

RaitoBezarius commented:
I do not know exactly how `retry` works, but basically whenever an exception is thrown, retry catch it and recall the function, right?

RaitoBezarius commented:
I think that for now, it should do it.

progval commented:
Yes. But let me remove the backoff increase first

RaitoBezarius commented:
Alright, let me know when I can merge the PR.

progval committed:
Don't increase backoff exponentially.

'retry' does not know about successes, so it will not reset the backoff after a success.

progval commented:
Done.

RaitoBezarius commented:
Thanks !
